OIDC: Support OAuth org to role mappings (#77683)
grafana/grafana: docs/sources/setup-grafana/configure-security/configure-authentication/github/index.md

You can disable this default role assignment by setting `role_attribute_strict = true`.
This setting denies user access if no role or an invalid role is returned.

> The `org_roles_attribute_path` configuration option can be used to add roles to other Orgs. See [Org roles mapping examples]

Reviewer: You can use the `org_roles_attribute_path` configuration option to add roles to other orgs. For more information, refer to [Org roles mapping examples]({{< relref "#org-roles-mapping-examples" >}}).

skip_org_role_sync = false
```

#### Org roles mapping examples

##### Map organization roles

> Given a user member of the `@my-github-organization/my-github-team1` and `@my-github-organization/my-github-team2` Gitub teams.

Reviewer 2: ```suggestion
Let's use the following Github teams: `@my-github-organization/my-github-team1` and `@my-github-organization/my-github-team2`.
```

Reviewer 2: You can also say "Let's say we have the following Github teams: `@my-github-organization/my-github-team1` and `@my-github-organization/my-github-team2`.

In this example, the user has been granted the role of a `Viewer` in the `team1` org, and the role of an `Editor` in the `team2` and `team3` orgs.

If the user was a member of the `@my-github-organization/admin` group, they would be granted the Grafana server administrator role.

Config:

```ini
role_attribute_path = contains(groups[*], '@my-github-organization/admin') && 'GrafanaAdmin' || 'None'
allow_assign_grafana_admin = true
org_attribute_path = groups
org_mapping = @my-github-organization/my-github-team1:team1:Viewer @my-github-organization/my-github-team2:team2:Editor *:team3:Editor
```

### Example of GitHub configuration in Grafana

> This section includes an example of GitHub configuration in the Grafana configuration file.

Reviewer 2: ```suggestion
Following is an example of a GitHub configuration in the Grafana configuration file.
```
